gammapy/cube/core.py

```diff
@@ -324,6 +320,10 @@ def solid_angle_image(self):
         sky_map = SkyMap.from_image_hdu(image_hdu)
         return sky_map.solid_angle()
 
+    def sky_image(self, slicepos):
```

Comment:
The version with the index is definitely needed.

E.g. when computing the solid angle image, one would simply do `sky_image(slicepos=0).solid_angle()` and when computing the PSF-convolved cube one would iterate over energy slice indices.

But yes, the use case of getting the slice for a given energy is also valid.
It could be added here by putting `energy_idx=None, energy=None` and then checking that the caller supplied exactly one.
Or we could rename [SkyCube.flux](https://gammapy.readthedocs.io/en/latest/api/gammapy.cube.SkyCube.html#gammapy.cube.SkyCube.flux) to `SkyCube.evaluate` and have a "neareast neighbor" option that doesn't do interpolation.

Maybe we table that question for now and get the basic version in first?